injector/network_disruption.go

```diff
@@ -89,6 +96,11 @@ type serviceWatcher struct {
 	servicesResourceVersion        string
 }
 
+type hostsWatcher struct {
+	// The only identifying info we need are the ip and filter priority
+	hostFilterMap map[v1beta1.NetworkDisruptionHostSpec][]*tcFilter
```

Comment:
I'll be quite honest that I wasn't sure if it's better to store a map of pointers to (slice of) pointers, structs to (slice of) pointers, or structs to (slice of) structs. I suppose these filter structs are only one `uint32` larger than their own pointer, so probably fine to store directly? The hostspecs should be okay to store directly, if i'm correctly understanding maps

Reply:
I would probably keep the simplest list to be able to reliably and consistently detect changes and changes only.

Then, if any changes, I would trigger a reinject.

Depending on the duration of such process (reinjection, that we will have to measure with the profiler or APM), this might be challenged to ensure the disruption is stable enough.